content/documentation/runtimes-monitoring/index.adoc

```diff
@@ -0,0 +1,228 @@
+---
+title: "Runtimes Monitoring"
+date: 2019-03-06T13:55:38+02:00
+draft: false
+type: "documentation"
+weight: 1
+---
+
+:linkattrs:
+
+= Monitoring your application runtimes
+:sectnums:
+:toc: left
+toc::[]
+:toc-title: Runtimes Monitoring
+:keywords: Kiali Documentation Runtimes Monitoring
+:icons: font
+:imagesdir: /images/documentation/runtimes-monitoring/
+
+Kiali can display custom dashboards to monitor application metrics. They are available for Workload and App.
+
+icon:bullhorn[size=2x]{nbsp} To show such dashboards, Kiali requires that the _app_ and _version_ labels are set on the pods. They are necessary for Kiali to identify which App and Workload they originate from.
+
+== Prometheus Configuration
+
+Kiali runtimes monitoring feature works exclusively with Prometheus, so it must be configured correctly to pull your application metrics.
+
+Hopefully in a standard Istio installation, Prometheus is already configured in such way, so you don't have to change anything and can jump to the next section.
+
+But if you want to use another Prometheus instance, please refer to link:https://prometheus.io/docs/prometheus/latest/configuration/configuration/#kubernetes_sd_config[Prometheus documentation] to setup `kubernetes_sd_config` for pods. As a reference, link:https://github.com/istio/istio/blob/907aa731c3f76ad21faac98614751e8ab3531893/install/kubernetes/helm/istio/charts/prometheus/templates/configmap.yaml#L229[here is] how it is configured in Istio.
+
+It is important to preserve labels mapping, so that Kiali can filter on App and Version. This is done with this config:
+
+```yaml
+      - action: labelmap
+        regex: __meta_kubernetes_pod_label_(.+)
+```
+
+Additionally, you have to tell Kiali what is the URL of that Prometheus server: either by setting the environment variable `PROMETHEUS_CUSTOM_METRICS_URL` or Kiali _ConfigMap_:
+
+```yaml
+# ...
+external_services:
+  prometheus_custom_metrics_url: URL_TO_SERVER
+# ...
+```
+
+Make sure that Kiali pod can reach this URL.
```

Comment:
Can we describe how to do that ?

Reply:
I don't want to open the pandora box :-) There can be so many different situations I guess.. prometheus in-cluster or not, behind a firewall or not, etc. I think at the moment the user has his own custom infrastructure, he is responsible for establishing the connections.